jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -829,7 +829,7 @@ ecma_builtin_array_prototype_object_slice (ecma_value_t arg1, /**< start */
 
   bool use_fast_path = ecma_op_object_is_fast_array (obj_p);
 #if ENABLED (JERRY_ES2015)
-  ecma_value_t new_array = ecma_op_array_species_create (obj_p, 0);
+  ecma_value_t new_array = ecma_op_array_species_create (obj_p, (uint32_t) JERRY_MAX ((int32_t) (end - start), 0));
```

Comment:
Could we factor out the new length into a const variable, and use that here, and later on instead of `length`? We should probably give it a more descriptive name as well, maybe somethng like `copied_length`.

After that, we could do an early return if `copied_length == 0` right after the result array is constructed. I believe this would allow us to omit `len != 0 && start < end` from the condition on line 853. (`len == 0` or `start >= end` would result in a copied length of zero, which would be handled by the early return.)

Reply:
Done.